lib/piwik/index.js

```diff
@@ -64,8 +67,35 @@ Piwik.prototype.track = function(track){
   var name = track.proxy('properties.name') || track.proxy('properties.label');
   var value = track.value() || track.revenue();
 
+  var options = track.options('Piwik');
+
+  // Add this if we enable setting your own customVariableLimit
+  //var customVariableLimit = this.options.customVariableLimit;
+
+  // Not currently enabling setting your own variable limit, setting this to the default
+  var customVariableLimit = 5;
+
+  var customVariables = options.customVars || options.cvar;
+  var formattedVars = {};
+
+  if (is.array(customVariables)) {
+    formattedVars = reduce(customVariables, function(formattedVars, value, i){
+      if (i <= customVariableLimit) formattedVars[i] = value;
+      return formattedVars;
+    }, {});
+  }
+
+  if (is.object(customVariables)) formattedVars = customVariables;
+
+  if (!is.null(formattedVars) && !is.undefined(formattedVars)) {
+    each(formattedVars, function(key, valuePair){
+      if (parseInt(key) <= customVariableLimit) push('setCustomVariable', key, valuePair[0], valuePair[1], 'page');
+    });
+  }
+
```

Comment:
You could simplify this logic if you converted objects into arrays, rather than the other way around:

```javascript
if (!is.object(customVariables) || !is.array(customVariables)) {
  customVariables = [];
}

if (is.object(customVariables)) {
  customVariables = objToArray(customVariables);
}

for (var i = 0; i < this.settings.customVariableLimit; i += 1){
  if (customVariables[i]) {
    push('setCustomVariable', i + 1, customVariableLimit[i][0], customVariableLimit[i][1], 'page');
  }
}
```